src/Backend/App.vala

```diff
@@ -159,16 +155,18 @@ public class Slingshot.Backend.App : Object {
                 count_visible = prop_value.get_boolean ();
             }
         }
-
-        unity_update_info ();
     }
 
     public void unity_reset () {
         unity_sender_name = null;
         count_visible = false;
         current_count = 0;
+    }
 
-        unity_update_info ();
+    public void remove_launcher_entry (string sender_name) {
```

Comment:
Okay then, I still disagree with this solution, but if this is how you think it should be, I won't be picking on it. If it's possible could we remove the entire `unity_reset ()` method and just put it's contents into here?